miner/worker_builder.go

```diff
@@ -438,25 +438,27 @@ func (w *worker) simulateBundle(
 			return nil, err
 		}
 
-		bundleGasUsed += receipt.GasUsed
+		if !w.eth.TxPool().Has(tx.Hash()) {
+			bundleGasUsed += receipt.GasUsed
 
-		txGasUsed := new(big.Int).SetUint64(receipt.GasUsed)
-		effectiveTip, err := tx.EffectiveGasTip(env.header.BaseFee)
-		if err != nil {
-			return nil, err
-		}
-		txGasFees := new(big.Int).Mul(txGasUsed, effectiveTip)
+			txGasUsed := new(big.Int).SetUint64(receipt.GasUsed)
+			effectiveTip, err := tx.EffectiveGasTip(env.header.BaseFee)
+			if err != nil {
+				return nil, err
+			}
+			txGasFees := new(big.Int).Mul(txGasUsed, effectiveTip)
 
-		if tx.Type() == types.BlobTxType {
-			blobFee := new(big.Int).SetUint64(receipt.BlobGasUsed)
-			blobFee.Mul(blobFee, receipt.BlobGasPrice)
-			txGasFees.Add(txGasFees, blobFee)
+			if tx.Type() == types.BlobTxType {
+				blobFee := new(big.Int).SetUint64(receipt.BlobGasUsed)
+				blobFee.Mul(blobFee, receipt.BlobGasPrice)
+				txGasFees.Add(txGasFees, blobFee)
+			}
+			bundleGasFees.Add(bundleGasFees, txGasFees)
+			sysBalanceAfter := state.GetBalance(consensus.SystemAddress)
+			sysDelta := new(uint256.Int).Sub(sysBalanceAfter, sysBalanceBefore)
+			sysDelta.Sub(sysDelta, uint256.MustFromBig(txGasFees))
+			ethSentToSystem.Add(ethSentToSystem, sysDelta.ToBig())
 		}
-		bundleGasFees.Add(bundleGasFees, txGasFees)
-		sysBalanceAfter := state.GetBalance(consensus.SystemAddress)
-		sysDelta := new(uint256.Int).Sub(sysBalanceAfter, sysBalanceBefore)
-		sysDelta.Sub(sysDelta, uint256.MustFromBig(txGasFees))
-		ethSentToSystem.Add(ethSentToSystem, sysDelta.ToBig())
 	}
 
 	bundleGasPrice := new(big.Int).Div(bundleGasFees, new(big.Int).SetUint64(bundleGasUsed))
```

Comment:
if all tx are seen in txpool, `bundleGasUsed` will be zero here, we can not div 0, please handle it.

Reply:
It is possible that the bundle is valid first and then become invalid as the tx become public in tx pool in future, and thus impact the whole block... how can we deal with it.